docs/PR_GUIDE.md

```diff
@@ -0,0 +1,56 @@
+# Pull Request Guide
+
+## General development rules:
+
+- All the changes of the code base are implemented through pull requests (PR) on master or other branches
+- When all the features for an upcoming release are implemented and are stable enough, a new release branch is made from master. After that, all app versions are updated in the master branch in files: ./omim/xcode/common.xcconfig and ./omim/android/gradle.properties
+- Only PR with bugfix should be done on release branches. If it’s necessary it’s permitted to cherry-pick such bugfix PRs to master
+- When a release is published one of developers makes a PR with all new commits of the release to master
+- All the further commits to release branch should be cherry-picked to the master by commits author
+- When a PR to any branch is about to merge, rebase and merge option should be used to make the code source tree flatter
+
+
+## Writing good PR:
+
+- PR should be small and reflect only one idea, a feature, a bugfix or a refactoring. In most cases a PR of 100 lines or less is considered as a PR of normal size and a PR of 1000 lines as big one
+- If a PR implements two different features, refactoring or bugfixes, it should be split into several PRs
+- A description field of every PR should contain a description, links to a ticket or/and links to article to confluence
+- If a PR implements a complex math task the description should contain a link to the proof of the solution or a link to a trusted source
+- All PRs should contain a corresponding ticket in jira. The exception is refactoring if there’s nothing to test after it. The reason for that is that every feature or bugfix should be tested by testers after it’s implemented
+- Every commit of all PRs should be compilable under all platforms. All tests should pass. So if changing of code breaks unit or integration tests these tests should be fixed in the same commit.
+- New functionality and unit or integration tests on it should be developed in the same PR
```

Comment:
in the same commit?

Reply:
I guess in the same PR. Imagine, you make some changes in one commit. All tests pass. Then you add some new tests on a new functionality.